e2e/kaivuilmoitus_muutosilmoitus.spec.ts

```diff
@@ -0,0 +1,455 @@
+  await page.getByLabel('Perustelut *').fill('Testiautomaatio e2e');
+  await page.getByLabel('Valitse päättäjä').getByText('Valitse päättäjä').click();
+  await page.getByText('Allu Päättäjä').click();
+  await page.getByRole('button', { name: 'TALLENNA' }).click();
+
+  await expect(page.getByLabel('Hakemus siirretty odottamaan')).toBeVisible();
+  await page.getByRole('button', { name: 'PÄÄTÄ' }).click();
+  await page.getByRole('button', { name: 'PÄÄTÄ' }).click();
+  await expect(page.getByLabel('Hakemus päätetty')).toBeVisible();
+
+  // tarkista haitattomasta
+  await page.goto(testiData.alluTriggerUrl);
+  await expect(async () => {
+    await page.goto(`${testiData.hankesalkku}${hanketunnus}`);
+    await page.getByText('Hakemukset').click();
+    await expect(page.getByTestId('application-status-tag').first()).toContainText('Päätös', {
+      timeout: 2000,
+    });
+    await expect(page.getByTestId('application-status-tag').nth(1)).toContainText('Päätös', {
+      timeout: 2000,
+    });
+  }).toPass({
+    intervals: [3000, 3000, 3000, 3000, 3000, 3000, 3000, 3000, 3000, 3000, 3000, 3000],
+    timeout: 360000,
+  });
+
+  // muutosilmoitus, lähetä valtakirja
+
+  await page.getByTestId(`applicationViewLinkIdentifier-${kaivuilmoitus}`).click();
+  await page.getByRole('button', { name: 'Tee muutosilmoitus' }).click();
+  await page
+    .locator('div')
+    .filter({ hasText: /^5Liitteet$/ })
+    .click();
+  await page
+    .locator('#excavation-notification-file-upload-mandate')
+    .setInputFiles(path.join(__dirname, './valtakirja_pdf.pdf'));
+  await expect(page.getByText('Lisätty tänään')).toBeVisible();
+  await page.getByRole('button', { name: 'Seuraava' }).click();
+  await expect(page.getByRole('heading', { name: 'Lisätty valtakirja' })).toBeVisible();
+  await expect(page.getByText('valtakirja_pdf.pdf')).toBeVisible();
+  await page.getByRole('button', { name: 'Lähetä muutosilmoitus' }).click();
+  await page.getByRole('button', { name: 'Vahvista' }).click();
+  await page.getByText('Muutosilmoitus lähetetty').click();
+
+  await page.goto(testiData.allu_url);
+  await page.getByPlaceholder('Username').fill(testiData.allupw);
+  await page.getByRole('button', { name: 'Submit' }).click();
+  await page.getByRole('link', { name: 'HAKEMUKSET' }).click();
+  await page.getByLabel('Hakemuksen tunnus').click();
+  await page.getByLabel('Hakemuksen tunnus').fill(`${kaivuilmoitus}`);
+  await page.getByRole('button', { name: 'HAE' }).click();
+  await expect(page.getByRole('link', { name: `${kaivuilmoitus}` })).toBeVisible();
+  await page.getByRole('link', { name: `${kaivuilmoitus}` }).click();
+  await page.getByRole('button', { name: 'KUITTAA' }).click();
+
+  await page.goto(`${testiData.hankesalkku}${hanketunnus}`);
+  await page.getByText('Hakemukset').click();
+  await page.getByTestId(`applicationViewLinkIdentifier-${kaivuilmoitus}`).click();
+
+  await expect(page.getByText('Hakemukselle on tehty')).toBeVisible();
```

Comment:
Should the test create a new decision in Allu from the muutosilmoitus? After all, a user creates a muutosilmoitus to get the updated decision with the changes they have reported.